theano/gof/link.py

```diff
@@ -172,27 +172,45 @@ def raise_with_op(node, thunk=None, exc_info=None, storage_map=None):
         # Prints output_map
         if storage_map is not None:
             detailed_err_msg += "\nStorage map footprint:\n"
+            storage_map_list = []
             for k in storage_map.keys():
                 if storage_map[k][0] is not None:
-                    detailed_err_msg += " - " + str(k) + ", "
+                    storage_map_item = []
+                    storage_map_item.append(str(k))
                     shapeinfo = None
                     if hasattr(storage_map[k][0], 'shape'):
                         shapeinfo = storage_map[k][0].shape
                         if len(shapeinfo) != 0:
```

Comment:
This was in the old code, but I just caught it. This if is bad. Can you remove it and remove the else portion? The if part should always be used. shapeinfo have a lenght of 0 for variable with ndim == 0. For them, we should not print (1, ) for there shape, but (). Otherwise, it make think they are a vector of lenght one and this isn't good.